src/opentelemetry.cr

```diff
@@ -110,13 +115,55 @@ module OpenTelemetry
   # ```
   def self.configure
     yield CONFIG
+
+    if ENV["OTEL_SERVICE_NAME"]?.presence && !CONFIG.service_name.presence
+      CONFIG.service_name = ENV["OTEL_SERVICE_NAME"]
+    end
+
+    # No need to configure the resource if already assigned within yield block
+    return unless CONFIG.resource.nil?
```

Comment:
I tested the resource creation on my honeycomb account with the block configuration, `OTEL_SERVICE_NAME` env variable, and `OTEL_RESOURCE_ATTRIBUTES` (including conflicting names with preference for the "former" over the "latter" options).

This made the configure method a bit long so all of this is supported. I'm totally fine with shuffling things around or extracting this new section into a `def self.configure_resource` of sorts to be called from here if preferred.